CompilerSource/compiler/components/write_defragged_events.cpp

```diff
@@ -89,20 +121,28 @@ int lang_CPP::compile_writeDefraggedEvents(const GameData &game, const ParsedObj
   /* Some events are included in all objects, even if the user
   ** hasn't specified code for them. Account for those here.
   ***********************************************************/
-  for (size_t i=0; i<event_sequence.size(); i++)
-  {
-    const int mid = event_sequence[i].first, id = event_sequence[i].second;
-    if (event_has_default_code(mid,id)) // We may not be using this event, but if it's persistent... (constant or defaulted: mandatory)
-    {
-      used_events[event_get_function_name(mid,id)].f2(mid,id); // ...Reserve it anyway.
+  for (const EventDescriptor &event : event_declarations()) {
+    // We may not be using this event, but it may have default code.
+    if (event.HasDefaultCode()) {  // (defaulted includes "constant")
+      // Defaulted events may NOT be parameterized.
+      used_events.insert(Event(event));
 
       for (parsed_object *obj : parsed_objects) { // Then shell it out into the other objects.
         bool exists = false;
-        for (unsigned j = 0; j < obj->events.size; j++)
-          if (obj->events[j].mainId == mid and obj->events[j].id == id)
-            { exists = true; break; }
-        if (!exists)
-          obj->events[obj->events.size] = parsed_event(mid, id, obj);
+        for (unsigned j = 0; j < obj->events.size; j++) {
+          Event ev2 = translate_legacy_id_pair(obj->events[j].mainId,
+                                               obj->events[j].id);
+          if (ev2.internal_id == event.internal_id) {
+            exists = true;
+            break;
+          }
+        }
+        if (!exists) {
+          std::cout << "EVENT SYSTEM: Adding a " << event.HumanName()
+                    << " event with default code.\n";
+          auto hack = reverse_lookup_legacy_event(event);
```

Comment:
Instead of just naming the variable "hack" you should probably put a proper FIXME: here with explanation in case someone other than you needs to fo fix this